### ISSUE TITLE
Add support for IngressClass introduced in Kubernetes v1.18

### ISSUE THREAD

**Issue by mikkeloscar** [zalando-incubator/kube-ingress-aws-controller#357]
Kubernetes v1.18 introduced a dedicated resource + field for defining `IngressClass`. See the documentation here: https://kubernetes.io/docs/concepts/services-networking/ingress/#ingress-class

It would make sense for the ingress-controller to support ingress class in this way going forward.

Related issue for skipper: https://github.com/zalando/skipper/issues/1478

**Comment by universam1** [comment]
Ran into this issue as well - apparently is not supported yet?

**Comment by mikkeloscar** [comment]
Yes, this is still not supported. You can still use the annotation if you need this: `kubernetes.io/ingress.class`
